the-collab-lab/tcl-68-smart-shopping-list
# Sm delete list

## Description

- User can now use delete button to remove the list they are currently on from firebase
- User is presented with a confirmation box upon clicking, with a warning that the action is not reversible
- List is also deleted from User's SharedLists, removing it from their home screen
- After successful deletion, user is navigated home and the list and items are removed from LocalStorage

## Related Issue

#40 -> As a user, I'd like to be able to delete a list.

## Acceptance Criteria

- [x] Add a button element to the `Manage List` page that upon click, deletes the selected list
- [x] When the user clicks the delete button, a confirmation pop up will ask the user to confirm if they want to delete the list
- [x] The list is successfully deleted from the users' lists, and the user is navigated back to the `Home` page

## Type of Changes

`Feature`

## Updates

ManageList.jsx, Firebase.jsx, App.jsx (added to props for ManageList route)

### Pictures of user facing dialog boxes:

<img width="452" alt="Screenshot 2024-03-28 at 9 10 16 AM" src="https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/117242156/b17de591-1a87-4d8a-aa85-d0c65ff2e27c">
<img width="451" alt="Screenshot 2024-03-28 at 9 11 00 AM" src="https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/117242156/6e95f83e-a20c-4a41-bfa5-17f1ce9a95db">
<img width="449" alt="Screenshot 2024-03-28 at 9 11 15 AM" src="https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/117242156/13d8f4b9-be08-40af-b3d4-7ff22db5cf2e">
<img width="452" alt="Screenshot 2024-03-28 at 10 15 46 AM" src="https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/117242156/9816ea8d-3660-458f-912a-523d74ad770e">

## Testing Steps / QA Criteria

1. Navigate to deployment below, or checkout branch `sm-delete-list`
2. Sign in, find three tests lists that have been shared, named Shared Test List, Shared Test List 2, Shared Test List 3
3. Select one of these test lists
4. Navigate to ManageList view, click 'DELETE this list'
5. Take note of warning, then:
5a. Click OK, and verify that the list is removed from Home, and the list has been unselected (no name visible, no items)
5b. Click Cancel, verify that no action is taken and the list persists in both your current list selection and on the Home view.

Note: This functionality meets the AC, but it should be known that 'orphaned' lists will be possible when a list is shared, but later deleted by the original creator. These lists will exist for individual users and we do not necessarily need to chase them down and remove them (as those users might still be using said lists after the originator wishes to be rid of them).
